src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -146,14 +172,30 @@ ss::future<> kafka_client_cache::clean_stale_clients() {
             return now >= (item.last_used + keep_alive);
         };
     };
+
+    auto units = _gc_lock.try_get_units();
+    if (!units) {
+        co_return;
+    }
+
+    gate_guard guard{_gc_gate};
+
     auto& inner_list = _cache.get<underlying_list>();
     co_await remove_client_if(inner_list, is_expired(_keep_alive));
```

Comment:
It's a reference to an index on `_cache`, it's around as long as the cache. `remove_client_if` is formulated in 2 stages; the first synchronous, and the second async, so that iterator invalidation can't happen.

Reply:
Inner_list is just an interface to allow for list-like modifications to the underlying container within boost_multi_index. The inner_list is not destroyed or anything like that, so yes it is still around.
Or maybe I'm not understanding your question?

Reply:
I see `inner_list` isn't even referenced across suspension points, this seems ok to me, resolving